minimal/server.go

```
@@ -190,6 +191,24 @@ func (t *txpoolHub) GetNonce(root types.Hash, addr types.Address) uint64 {
 	return account.Nonce
 }
 
+func (t *txpoolHub) GetBalance(root types.Hash, addr types.Address) *big.Int {
+	zeroValue := big.NewInt(0)
+
+	snap, err := t.state.NewSnapshotAt(root)
+	if err != nil {
+		return zeroValue
+	}
+	result, ok := snap.Get(keccak.Keccak256(nil, addr.Bytes()))
+	if !ok {
+		return zeroValue
+	}
+	var account state.Account
+	if err = account.UnmarshalRlp(result); err != nil {
+		return zeroValue
```

I think we should return an error here if we cannot unmarshal the account.
The only time where returning 0 makes sense IMO is when we cannot find the account in the snapshot that we know is there.

---

Agreed.

Fixed 👍 

2771012